subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcApiChangePerformanceTest.groovy

```diff
@@ -61,6 +59,40 @@ class BuildSrcApiChangePerformanceTest extends AbstractCrossVersionPerformanceTe
         result.assertCurrentVersionHasNotRegressed()
     }
 
+    def "buildSrc non-abi change"() {
+        given:
+        runner.tasksToRun = ['help']
+        runner.runs = determineNumberOfRuns(runner.testProject)
+
+        and:
+        runner.addBuildMutator { settings ->
+            def changingClassFilePath = new File(settings.projectDir, 'buildSrc/src/main/groovy/ChangingClass.groovy')
+            new BuildMutator() {
+                @Override
+                void beforeScenario(ScenarioContext context) {
+                    writeFile(changingClassFilePath, """
+                        class ChangingClass {
+                            void changingMethod${context.phase}${context.iteration}() {
```

Comment:
```suggestion
                            void changingMethod${context.uniqueBuildId}() {
```

It is better to use `context.uniqueBuildId`, since then the name is unique to the actual build and can never be from the cache.

Note that you are generating the class in `beforeScenario`, so you could as well use a non-changing method name.

Reply:
It was a copy-paste from the existing test. I've removed the dynamic part of the method name.